src/OpenTK.Mathematics/Matrix/Matrix2.cs

```diff
@@ -551,7 +551,7 @@ public static Matrix2 Subtract(Matrix2 left, Matrix2 right)
         /// <exception cref="InvalidOperationException">Thrown if the Matrix2 is singular.</exception>
         public static void Invert(in Matrix2 mat, out Matrix2 result)
         {
-            var det = mat.Determinant;
+            var det = (mat.Row0.X * mat.Row1.Y) - (mat.Row0.Y * mat.Row1.X);
```

Comment:
Is the reason for this to avoid shadow copy? If so it would probably be better to give `Determinant` a `readonly` modifier.

Reply:
The reason is to manually inline this calculation as the JIT doesn't do it. I've looked at the disassembly output for this function and this property access does not get inlined and manually inlining it improves performance.